staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go

```diff
@@ -90,6 +90,32 @@ type APIServiceSpec struct {
 
 	// leaving this here so everyone remembers why proto index 6 is skipped
 	// Priority int64 `json:"priority" protobuf:"varint,6,opt,name=priority"`
+
+	// ScopedResources declares a range of scoped resources to tell what resources to provide by the APIService. there can be more resources provided
+	// by the servicebackend. k8s aggregator focuses on the resources declared by ScopedResources, if it is defined.
+	ScopedResources []ScopedResource `json:"scopedResources,omitempty" protobuf:"bytes,7,opt,name=scopedResources"`
+}
+
+type ResourceScope string
+
+const (
```

Comment:
my understanding is sparse, but i feel that new fields cannot be added to stable versions, particularly one that causes change in behavior.

Reply:
I learnt that my statement was not fully accurate. New fields can be added under alpha flag

example for how Immutable was added to stable version:
https://github.com/kubernetes/kubernetes/pull/86377 
https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/1412-immutable-secrets-and-configmaps/README.md

Reply:
agree with @AryanSharma9917 that new fields goes to v1beta1 or v1alpha1 version first before it goes mature.